### PR TITLE
pdf convertor

### LINKED ISSUE #2: Consistent rendering for PDF files across rendering software
## Description
The same PDF file created using puppeteer renders differently across rendering software.

### Apple Preview
Screenshot 2023-09-21 at 2 57 11 PM

### Google Chrome

Screenshot 2023-09-21 at 2 57 38 PM

## Suggested Fix

Using a commandline tool like [Ghostscript](https://www.ghostscript.com/index.html) to convert the PDF to PDF/A seems to solve for this problem.

### Example Script
```bash
gs -dPDFA -dBATCH -dNOPAUSE -sProcessColorModel=DeviceCMYK -sDEVICE=pdfwrite -sPDFACompatibilityPolicy=1 -sOutputFile=output.pdf input.pdf
```

### PR DESCRIPTION
**Pull request by KDwevedi**
fixes #2 

Changes:
- Added install command to Dockerfile for [ghostscript](https://www.ghostscript.com/) to use it's commandline utility to convert pdfs to pdf/a format
- Created a new service to execute the conversion
- modified rcw/credential to call the new service before uploading to minio

### CODE REVIEW

File: src/rcw/pdf.service.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import { promisify } from 'util';
+
+const exec = promisify(execCb);
+
+@Injectable()
+export class ExecService {
+  async pdfConvertorCommand(filePath, outputPath): Promise<void> {
+    try {
+      const { stdout, stderr } = await exec(`gs -dPDFA -dBATCH -dNOPAUSE -sProcessColorModel=DeviceCMYK -sDEVICE=pdfwrite -sPDFACompatibilityPolicy=1 -sOutputFile="${outputPath}" "${filePath}"`);
+      console.log(`stdout: ${stdout}`);
```
Comment (techsavvyash) [inline src/rcw/pdf.service.ts]: Let's move these logs to nestJS's built in logger.

File: src/rcw/rcw.service.ts [hunk 2]
```diff
@@ -715,6 +722,7 @@ export class RcwService {
     // delete the pdf file to freee up storage
     try {
       await fs.promises.unlink(filePath);
+      await fs.promises.unlink(outputPath);
```
Comment (techsavvyash) [inline src/rcw/rcw.service.ts]: since we are deleting multiple files now, there should be some faster way to combine this statement and one above it.

**Review by techsavvyash — CHANGES_REQUESTED** [review]

File: src/rcw/pdf.service.ts (hunk 1 above)
Comment (KDwevedi) [reply src/rcw/pdf.service.ts]: Done

File: src/rcw/rcw.service.ts (hunk 2 above)
Comment (KDwevedi) [reply src/rcw/rcw.service.ts]: Updated to use Promises.all

**Pull request merged**